File: Archivos/Ape05/BancoMain.cpp
// Librerías necesarias
#include <iostream>
#include <fstream>
#include "Cuenta.h"

using namespace std;

int main() {

    // Crear objeto cuenta
    Cuenta cuenta;

    // Variables
    string usuario;
    double saldoInicial;
    double monto;
    int opcion;

    // Archivo para guardar datos
    ofstream archivo("cajero.txt");

    // Ingreso de datos
    cout << "Ingrese el nombre del usuario: ";
    getline(cin, usuario);

    cout << "Ingrese el saldo inicial: ";
    cin >> saldoInicial;

    // Guardar datos en el objeto
    cuenta.setUsuario(usuario);
    cuenta.setSaldo(saldoInicial);

    // Menú principal
    do {

        cout << "\n======= CAJERO AUTOMATICO =======" << endl;
        cout << "1. Consultar saldo" << endl;
        cout << "2. Depositar dinero" << endl;
        cout << "3. Retirar dinero" << endl;
        cout << "4. Salir" << endl;
        cout << "Seleccione una opcion: ";
        cin >> opcion;

        switch(opcion) {

            case 1:

                // Mostrar saldo
                cuenta.consultarSaldo();
                break;

            case 2:

                // Depositar dinero
                cout << "Ingrese el monto a depositar: ";
                cin >> monto;

                cuenta.depositar(monto);

                cout << "Deposito realizado correctamente" << endl;
                break;

            case 3:

                // Retirar dinero
                cout << "Ingrese el monto a retirar: ";
                cin >> monto;

                cuenta.retirar(monto);
                break;

            case 4:

                cout << "Saliendo del sistema..." << endl;
                break;

            default:

                cout << "Opcion invalida" << endl;
        }

    } while(opcion != 4);

    // Guardar datos en archivo
    archivo << "Usuario: " << cuenta.getUsuario() << endl;
    archivo << "Saldo final: $" << cuenta.getSaldo() << endl;

    // Cerrar archivo
    archivo.close();

    return 0;
}

File: Archivos/Ape05/BancoMain.java
// Librerías necesarias
import java.util.Scanner;
import java.io.FileWriter;
import java.io.IOException;

public class BancoMain{

    public static void main(String[] args) {

        // Objeto Scanner
        Scanner entrada = new Scanner(System.in);

        // Crear objeto cuenta
        Cuenta cuenta = new Cuenta();

        // Variables
        String usuario;
        double saldoInicial;
        double monto;
        int opcion;

        try {

            // Archivo para guardar datos
            FileWriter archivo = new FileWriter("cajerojava.txt");

            // Ingreso de datos
            System.out.print("Ingrese el nombre del usuario: ");
            usuario = entrada.nextLine();

            System.out.print("Ingrese el saldo inicial: ");
            saldoInicial = entrada.nextDouble();

            // Guardar datos en el objeto
            cuenta.setUsuario(usuario);
            cuenta.setSaldo(saldoInicial);

            // Menú principal
            do {

                System.out.println("\n======= CAJERO AUTOMATICO =======");
                System.out.println("1. Consultar saldo");
                System.out.println("2. Depositar dinero");
                System.out.println("3. Retirar dinero");
                System.out.println("4. Salir");
                System.out.print("Seleccione una opcion: ");
                opcion = entrada.nextInt();

                switch(opcion) {

                    case 1:

                        // Consultar saldo
                        cuenta.consultarSaldo();
                        break;

                    case 2:

                        // Depositar dinero
                        System.out.print("Ingrese el monto a depositar: ");
                        monto = entrada.nextDouble();

                        cuenta.depositar(monto);

                        System.out.println("Deposito realizado correctamente");
                        break;

                    case 3:

                        // Retirar dinero
                        System.out.print("Ingrese el monto a retirar: ");
                        monto = entrada.nextDouble();

                        cuenta.retirar(monto);
                        break;

                    case 4:

                        System.out.println("Saliendo del sistema...");
                        break;

                    default:

                        System.out.println("Opcion invalida");
                }

            } while(opcion != 4);

            // Guardar datos en archivo
            archivo.write("Usuario: " + cuenta.getUsuario() + "\n");
            archivo.write("Saldo final: $" + cuenta.getSaldo() + "\n");

            // Cerrar archivo
            archivo.close();

        } catch(IOException e) {

            System.out.println("Error al crear el archivo");
        }

        // Cerrar Scanner
        entrada.close();
    }
}
